src/server/game/Entities/Item/ItemTemplate.cpp

```diff
@@ -122,3 +126,152 @@ uint32 ItemTemplate::GetSkill() const
+        QueryData[loc] = queryTemp;
+    }
+}
+
+WorldPacket ItemTemplate::BuildQueryData(LocaleConstant loc) const
+{
+    WorldPacket queryTemp(SMSG_ITEM_QUERY_SINGLE_RESPONSE, 500);
+
+    std::string locName = Name1;
+    std::string locDescription = Description;
+
+    if (ItemLocale const* il = sObjectMgr->GetItemLocale(ItemId))
+    {
+        ObjectMgr::GetLocaleString(il->Name, loc, locName);
+        ObjectMgr::GetLocaleString(il->Description, loc, locDescription);
+    }
+
+    queryTemp << ItemId;
+    queryTemp << Class;
+    queryTemp << SubClass;
+    queryTemp << SoundOverrideSubclass;
+    queryTemp << locName;
+    queryTemp << uint8(0x00);                                //Name2; // blizz not send name there, just uint8(0x00); <-- \0 = empty string = empty name...
+    queryTemp << uint8(0x00);                                //Name3; // blizz not send name there, just uint8(0x00);
+    queryTemp << uint8(0x00);                                //Name4; // blizz not send name there, just uint8(0x00);
+    queryTemp << DisplayInfoID;
+    queryTemp << Quality;
+    queryTemp << Flags;
+    queryTemp << Flags2;
+    queryTemp << BuyPrice;
+    queryTemp << SellPrice;
+    queryTemp << InventoryType;
+    queryTemp << AllowableClass;
+    queryTemp << AllowableRace;
+    queryTemp << ItemLevel;
+    queryTemp << RequiredLevel;
+    queryTemp << RequiredSkill;
+    queryTemp << RequiredSkillRank;
+    queryTemp << RequiredSpell;
+    queryTemp << RequiredHonorRank;
+    queryTemp << RequiredCityRank;
+    queryTemp << RequiredReputationFaction;
+    queryTemp << RequiredReputationRank;
+    queryTemp << int32(MaxCount);
+    queryTemp << int32(Stackable);
+    queryTemp << ContainerSlots;
+    queryTemp << StatsCount;                         // item stats count
+    for (uint32 i = 0; i < StatsCount; ++i)
+    {
+        queryTemp << ItemStat[i].ItemStatType;
+        queryTemp << ItemStat[i].ItemStatValue;
+    }
+    queryTemp << ScalingStatDistribution;            // scaling stats distribution
+    queryTemp << ScalingStatValue;                   // some kind of flags used to determine stat values column
+    for (int i = 0; i < MAX_ITEM_PROTO_DAMAGES; ++i)
+    {
+        queryTemp << Damage[i].DamageMin;
+        queryTemp << Damage[i].DamageMax;
+        queryTemp << Damage[i].DamageType;
+    }
+
+    // resistances (7)
+    queryTemp << Armor;
+    queryTemp << HolyRes;
+    queryTemp << FireRes;
+    queryTemp << NatureRes;
+    queryTemp << FrostRes;
+    queryTemp << ShadowRes;
+    queryTemp << ArcaneRes;
+
+    queryTemp << Delay;
+    queryTemp << AmmoType;
+    queryTemp << RangedModRange;
+
+    for (int s = 0; s < MAX_ITEM_PROTO_SPELLS; ++s)
+    {
+        // send DBC data for cooldowns in same way as it used in Spell::SendSpellCooldown
+        // use `item_template` or if not set then only use spell cooldowns
+        SpellInfo const* spell = sSpellMgr->GetSpellInfo(Spells[s].SpellId);
+        if (spell)
```

Comment:
this shit is wrong and was wrong before you changed it too, only db data should ever be sent